@nazar-pc I've tried everything here but definitely I don't understand why this fails:

https://github.com/versatica/mediasoup/pull/1015/commits/e3f425a5924a17a0016704653c0c600f355234d9

It complains:

```
$ KEEP_BUILD_ARTIFACTS=1 cargo test
   Compiling mediasoup v0.11.4 (/Users/ibc/src/v3-mediasoup/rust)
error[E0308]: mismatched types
   --> rust/src/ortc.rs:673:42
    |
673 |               *codec.rtcp_feedback_mut() = matched_cap_codec
    |  _____________--------------------------___^
    | |             |
    | |             expected due to the type of this binding
674 | |                 .rtcp_feedback().clone()
675 | |                 .retain(|fb| enable_rtx || fb != &RtcpFeedback::Nack);
    | |_____________________________________________________________________^ expected struct `Vec`, found `()`
    |
    = note: expected struct `Vec<RtcpFeedback>`
            found unit type `()`
note: method `retain` modifies its receiver in-place
   --> rust/src/ortc.rs:675:18
    |
673 |               *codec.rtcp_feedback_mut() = matched_cap_codec
    |  __________________________________________-
674 | |                 .rtcp_feedback().clone()
    | |________________________________________- you probably want to use this value after calling the method...
675 |                   .retain(|fb| enable_rtx || fb != &RtcpFeedback::Nack);
    |                    ^^^^^^ this call modifies its receiver in-place
    = note: ...instead of the `()` output of method `retain`

error[E0308]: mismatched types
   --> rust/src/ortc.rs:673:42
    |
673 |               *codec.rtcp_feedback_mut() = matched_cap_codec
    |  _____________--------------------------___^
    | |             |
    | |             expected due to the type of this binding
674 | |                 .rtcp_feedback().clone()
675 | |                 .retain(|fb| enable_rtx || fb != &RtcpFeedback::Nack);
    | |_____________________________________________________________________^ expected struct `Vec`, found `()`
    |
    = note: expected struct `Vec<rtp_parameters::RtcpFeedback>`
            found unit type `()`
note: method `retain` modifies its receiver in-place
   --> rust/src/ortc.rs:675:18
    |
673 |               *codec.rtcp_feedback_mut() = matched_cap_codec
    |  __________________________________________-
674 | |                 .rtcp_feedback().clone()
    | |________________________________________- you probably want to use this value after calling the method...
675 |                   .retain(|fb| enable_rtx || fb != &RtcpFeedback::Nack);
    |                    ^^^^^^ this call modifies its receiver in-place
    = note: ...instead of the `()` output of method `retain`
```

I **do** know that `retain()` modifies its receiver in place, that's why I'm doing `clone()` first, but it doesn't care it seems...

Regarding first error " expected struct `Vec`, found `()`", absolutely no idea what it means.

## UPDATE

I've got a solution that works, but not sure how good/idiomatic it is:

### Code in v3

```rust
*codec.rtcp_feedback_mut() = matched_cap_codec.rtcp_feedback().clone();
```

### Desired code in this branch (it doesn't compile as shown above)

```rust
*codec.rtcp_feedback_mut() = matched_cap_codec
    .rtcp_feedback().clone()
    .retain(|fb| enable_rtx || fb != &RtcpFeedback::Nack);
```

### Working code

```rust
*codec.rtcp_feedback_mut() = matched_cap_codec.rtcp_feedback().clone();

codec
    .rtcp_feedback_mut()
    .retain(|fb| enable_rtx || fb != &RtcpFeedback::Nack);
```

It compiles, no warnings, and makes the new test pass. Should I go this way?
NOTE: I'm gonna push it since it's way better than the failing thing. Pushed here https://github.com/versatica/mediasoup/pull/1015/commits/a84ab48d923703f991d22f36fd6b22fda9992518

--- a/node/src/Transport.ts
+++ b/node/src/Transport.ts
@@ -694,14 +694,7 @@ export class Transport<Events extends TransportEvents = TransportEvents,
 		// If enableRtx is not given, set it to true if video and false if audio.
 		if (enableRtx === undefined)
 		{
-			if (producer.kind === 'video')
-			{
-				enableRtx = true;
-			}
-			else
-			{
-				enableRtx = false;
-			}
+			enableRtx = producer.kind === 'video';
 		}
 
 		// This may throw.

--- a/rust/src/ortc.rs
+++ b/rust/src/ortc.rs
@@ -668,8 +668,10 @@ pub(crate) fn get_consumer_rtp_parameters(
             .iter()
             .find(|cap_codec| match_codecs(cap_codec.deref().into(), (&codec).into(), true).is_ok())
         {
+            // TODO: Remove this.
             // *codec.rtcp_feedback_mut() = matched_cap_codec.rtcp_feedback().clone();
 
+            // TODO: This doesn't compile. Fix it.
             *codec.rtcp_feedback_mut() = matched_cap_codec
                 .rtcp_feedback().clone()
                 .retain(|fb| enable_rtx || fb != &RtcpFeedback::Nack);
@@ -745,16 +747,6 @@ pub(crate) fn get_consumer_rtp_parameters(
             codec
                 .rtcp_feedback_mut()
                 .retain(|fb| !matches!(fb, RtcpFeedback::GoogRemb | RtcpFeedback::TransportCc));
-        }
-    }
-
-    // If enable_rtx is false, remove RTCP NACK support. If so, we remove all
-    // RTCP feedbacks with type 'nack' (including those whose parameter is 'pli').
-    if !enable_rtx {
-        for codec in &mut consumer_params.codecs {
-            codec
-                .rtcp_feedback_mut()
-                .retain(|fb| fb != &RtcpFeedback::Nack);
         }
     }
 

--- a/rust/src/ortc.rs
+++ b/rust/src/ortc.rs
@@ -668,13 +668,9 @@ pub(crate) fn get_consumer_rtp_parameters(
             .iter()
             .find(|cap_codec| match_codecs(cap_codec.deref().into(), (&codec).into(), true).is_ok())
         {
-            // TODO: Remove this.
-            // *codec.rtcp_feedback_mut() = matched_cap_codec.rtcp_feedback().clone();
+            *codec.rtcp_feedback_mut() = matched_cap_codec.rtcp_feedback().clone();
 
-            // TODO: This doesn't compile. Fix it.
-            *codec.rtcp_feedback_mut() = matched_cap_codec
-                .rtcp_feedback().clone()
-                .retain(|fb| enable_rtx || fb != &RtcpFeedback::Nack);
+            codec.rtcp_feedback_mut().retain(|fb| enable_rtx || fb != &RtcpFeedback::Nack);
 
             consumer_params.codecs.push(codec);
         }

